packages/dev/core/src/Engines/WebGPU/webgpuTextureManager.ts

```diff
@@ -379,19 +363,60 @@ export class WebGPUTextureManager {
 
             let modules = this._compiledShaders[index];
             if (!modules) {
-                let vertexCode = this._glslang.compileGLSL(defines + shadersForPipelineType[type].vertex, "vertex");
-                let fragmentCode = this._glslang.compileGLSL(defines + shadersForPipelineType[type].fragment, "fragment");
+                let vertexCode = shadersForPipelineType[type].vertex;
+                let fragmentCode = shadersForPipelineType[type].fragment;
+
+                const processorOptions: ProcessingOptions = {

```

Comment:
I wonder if we should not factor somewhere the code that compiles shader (we have such code here and in `Effect`)...

Reply:
We have a few differences unfortunately. We can consider it in a new PR though.